src/SignalSlot/RemoveTranslationSlot.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\PlatformHttpCacheBundle\SignalSlot;
+
+use EzSystems\PlatformHttpCacheBundle\PurgeClient\PurgeClientInterface;
+use eZ\Publish\Core\SignalSlot\Signal;
+use eZ\Publish\SPI\Persistence\Handler as PersistenceHandler;
+
+class RemoveTranslationSlot extends AbstractContentSlot
+{
+    /**
+     * @var \eZ\Publish\SPI\Persistence\Handler
+     */
+    protected $persistenceHandler;
+
+    /**
+     * @param \EzSystems\PlatformHttpCacheBundle\PurgeClient\PurgeClientInterface $purgeClient
+     * @param \eZ\Publish\SPI\Persistence\Handler $persistenceHandler
+     */
+    public function __construct(
+        PurgeClientInterface $purgeClient,
+        PersistenceHandler $persistenceHandler
+    ) {
+        parent::__construct($purgeClient);
+        $this->persistenceHandler = $persistenceHandler;
+    }
+
+    /**
+     * Checks if $signal is supported by this handler.
+     *
+     * @param \eZ\Publish\Core\SignalSlot\Signal $signal
+     *
+     * @return bool
+     */
+    protected function supports(Signal $signal)
+    {
+        return $signal instanceof Signal\ContentService\RemoveTranslationSignal;
+    }
+
+    /**
+     * Generate tags for content, relation, locations and parent locations.
+     *
+     * RemoveTranslationSignal doesn't provide locationId, so Locations need to be fetched
+     *
+     * @param \eZ\Publish\Core\SignalSlot\Signal $signal
+     *
+     * @return array
+     */
+    protected function generateTags(Signal $signal)
+    {
+        $tags = [];
+        if (isset($signal->contentId)) {
+            // self in all forms (also without locations)
+            $tags[] = 'content-' . $signal->contentId;
+            // reverse relations
+            $tags[] = 'relation-' . $signal->contentId;
+        }
+
+        $locations = $this->persistenceHandler->locationHandler()->loadLocationsByContent(
+            $signal->contentId
+        );
+        foreach ($locations as $location) {
+            $tags[] = 'location-' . $location->id;
```

Comment:
> why do you need to clear location id? you can just clear by content id, and it will clear all relevant cache items. This is one of the many points with multi tagging, you don't need to always refer to cache by location id.

Interesting thing, the moment I comment-out this line, it stopped working returning cached version (checked using [this test](https://github.com/ezsystems/ezpublish-kernel/pull/2065/commits/011206b414f2cbae600533f069701da841159c54)). Don't really know anything about tagging to tell you why :(

> But ok for parent/siblings I see the point, but unsure if it is needed.

However I can't think of - looking at [REST API v2 spec](https://github.com/ezsystems/ezpublish-kernel/blob/master/doc/specifications/rest/REST-API-V2.rst) - any case when we can access Version via parent location. That would be a good case to check if `parent-` tag is needed there.